oci/oci.go

```diff
@@ -112,31 +125,69 @@ func (r *Runtime) Name() string {
 	return r.name
 }
 
+// Runtimes returns the map of OCI runtimes.
+func (r *Runtime) Runtimes() map[string]RuntimeHandler {
+	return r.runtimes
+}
+
+// ValidateRuntimeHandler returns an error if the runtime handler string
+// provided does not match any valid use case.
+func (r *Runtime) ValidateRuntimeHandler(handler string) (RuntimeHandler, error) {
+	if handler == "" {
+		return RuntimeHandler{}, fmt.Errorf("empty runtime handler")
+	}
+
+	runtimeHandler, ok := r.runtimes[handler]
+	if !ok {
+		if handler == UntrustedRuntime && r.untrustedPath != "" {
+			return RuntimeHandler{
+				RuntimePath: r.untrustedPath,
+			}, nil
+		}
+		return RuntimeHandler{}, fmt.Errorf("failed to find runtime handler %s from runtime list %v",
+			handler, r.runtimes)
+	}
+	if runtimeHandler.RuntimePath == "" {
+		return RuntimeHandler{}, fmt.Errorf("empty runtime path for runtime handler %s", handler)
+	}
+
+	return runtimeHandler, nil
+}
+
 // Path returns the full path the OCI Runtime executable.
 // Depending if the container is privileged and/or trusted,
 // this will return either the trusted or untrusted runtime path.
-func (r *Runtime) Path(c *Container) string {
+func (r *Runtime) Path(c *Container) (string, error) {
+	if c.runtimeHandler != "" {
+		runtimeHandler, err := r.ValidateRuntimeHandler(c.runtimeHandler)
+		if err != nil {
+			return "", err
+		}
+
+		return runtimeHandler.RuntimePath, nil
+	}
+
 	if !c.trusted {
 		if r.untrustedPath != "" {
-			return r.untrustedPath
+			return r.untrustedPath, nil
 		}
 
-		return r.trustedPath
+		return r.trustedPath, nil
```

Comment:
trustedPath is basically whatever we have in `Runtime`, I think we need to always prefer Runtimes with a default one

Reply:
I believe this is taken care of by the code above this one rigtht? @sboeuf

Reply:
Yes the code above this will first try to handle a `runtime_handler` if provided.